test/bddtests/did_sidetree_steps.go

```diff
@@ -99,29 +97,44 @@ func (d *DIDSideSteps) resolveDIDDocument() error {
 	if err != nil {
 		return err
 	}
-	d.resp, err = restclient.SendResolveRequest(testDocumentURL + "/" + documentHash)
-	return err
+
+	remainingAttempts := 20
+	for {
+		d.resp, err = restclient.SendResolveRequest(testDocumentURL + "/" + documentHash)
+		if err != nil {
+			return err
+		}
+		if d.resp.StatusCode == http.StatusNotFound {
+			logger.Infof("Document not found: %s. Remaining attempts: %d", documentHash, remainingAttempts)
+			remainingAttempts--
+			if remainingAttempts > 0 {
+				time.Sleep(time.Second)
+				continue
+			}
```

Comment:
we should return an error if we exhausted attempts

Reply:
I didn't want to create a new error to return since we're returning the 404 - 'error not found' message.

Reply:
sounds good